napari/utils/colormaps/colormap.py

```diff
@@ -230,3 +235,62 @@ def default_color(self):
         return self.color_dict.get(None, (0, 0, 0, 0))
         # we provided here default color for backward compatibility
         # if someone is using DirectLabelColormap directly, not through Label layer
+
+
+def cast_labels_to_minimum_type_auto(
+    data: np.ndarray, num_colors: int
+) -> np.ndarray:
+    """Perform modulo operation based on number of colors
+
+    Parameters
+    ----------
+    data : np.ndarray
+        Labels data to be casted.
+    num_colors : int
+        Number of unique colors in the data.
+
+    Returns
+    -------
+    np.ndarray
+        Casted labels data.
+    """
+    dtype = minimum_dtype_for_labels(num_colors + 1)
+
+    return _cast_labels_to_minimum_type_auto(data, num_colors, dtype)
+
+
+@numba.njit(parallel=True)
+def _cast_labels_to_minimum_type_auto(
+    data: np.ndarray, num_colors: int, dtype
+) -> np.ndarray:
```

Comment:
I think we might as well make this function have a generic style here already?

```suggestion
def modulo_plus_one(
    values: np.ndarray, n: int, dtype
) -> np.ndarray:
    """Like ``array % n`` but with 1 added to result for values >n.

    This ensures (1) an output value in [0, n] (inclusive), and (2) that
    no nonzero values in the input are zero in the output.
    """
```

Reply:
This is intentionally private function for hiding implementation details. 

Public API is in `cast_labels_to_minimum_type_auto`. Using this directly may be not a proper idea, as I do not add any guard about dtype.

Reply:
It can be private, but I think it's useful to rename it as above and add a proper docstring, as it makes it easier to understand the logic.